pkg/cfn/manager/nodegroup.go

```diff
@@ -74,6 +74,45 @@ func (c *StackCollection) createManagedNodeGroupTask(errorCh chan error, ng *api
 	return c.CreateStack(name, stack, ng.Tags, nil, errorCh)
 }
 
+func (c *StackCollection) propagateManagedNodeGroupTagsToASGTask(errorCh chan error, ng *api.ManagedNodeGroup) error {
+	// describe node group to retrieve ASG names
+	input := &eks.DescribeNodegroupInput{
+		ClusterName:   aws.String(c.spec.Metadata.Name),
+		NodegroupName: aws.String(ng.Name),
+	}
+	res, err := c.eksAPI.DescribeNodegroup(input)
+	if err != nil {
+		return errors.Wrapf(err, "couldn't get managed nodegroup details for nodegroup %q", ng.Name)
+	}
+
+	// set the managed nodegroup tags to all the ASGs found
+	if res.Nodegroup.Resources != nil {
+		// build the input tags for all ASGs attached to the managed nodegroup
+		asgTags := []*autoscaling.Tag{}
+
+		for _, asg := range res.Nodegroup.Resources.AutoScalingGroups {
```

Comment:
Please add a check if the number of tags on the ASG would go over the limit of 50 and throw an error in that case.

Reply:
https://docs.aws.amazon.com/autoscaling/ec2/userguide/autoscaling-tagging.html#tag_restrictions @SlevinWasAlreadyTaken

Reply:
Done in 0099c487ccebe4d8141505c8db17fcaa3f158ff0
Checked both the 50 limit and the 25 (creating tags by chunk of 25 tags if necessary)